telemetry/csharp/ToolkitTelemetryGenerator/DefinitionsBuilder.cs

```diff
@@ -0,0 +1,757 @@
+﻿using System;
+using System.CodeDom;
+using System.CodeDom.Compiler;
+using System.Collections.Generic;
+using System.Globalization;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+using ToolkitTelemetryGenerator.Models;
+using ToolkitTelemetryGenerator.Utils;
+
+namespace ToolkitTelemetryGenerator
+{
+    /// <summary>
+    /// Generates code that allows programs like the Toolkit to instantiate and publish Telemetry Events
+    /// </summary>
+    public class DefinitionsBuilder
+    {
+        private const string MetadataEntryFullName = "Amazon.ToolkitTelemetry.Model.MetadataEntry";
+        private const string MetricDatumFullName = "Amazon.ToolkitTelemetry.Model.MetricDatum";
+        private const string AddMetadataMethodName = "AddMetadata";
+        private readonly List<MetricType> _types = new List<MetricType>();
+        private readonly List<Metric> _metrics = new List<Metric>();
+
+        private readonly CodeMethodReferenceExpression _invariantCulture =
+            new CodeMethodReferenceExpression(new CodeTypeReferenceExpression(typeof(CultureInfo)),
+                nameof(CultureInfo.InvariantCulture));
+
+        private string _namespace;
+
+        private CodeCompileUnit _generatedCode;
```

Comment:
What's the benefit of defining these out here - but then only instantiating and using them as part of the `Build` method? Could they not be local to that method?

Reply:
Nice catch, sometimes the code at the start of a journey gets overlooked when tidying up at the destination